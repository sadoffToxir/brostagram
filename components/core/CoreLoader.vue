<template>
  <v-overlay
    :model-value="storeLoader.isLoading"
    class="align-center justify-center"
    z-index="999"
    persistent
  >
    <v-progress-circular
      color="primary"
      indeterminate
      size="64"
      :data-test="componentTag"
    />
  </v-overlay>
</template>

<script setup lang="ts">

const componentTag = 'CoreLoader'
const storeLoader = useLoader()
</script>
