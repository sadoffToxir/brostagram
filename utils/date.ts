type CustomDate = string | number | Date

export const formatToIsoDate = (date: CustomDate) => new Date(date).toISOString().split('T')[0]
export const formatToIsoDateAndTime = (date: CustomDate) => new Date(date).toISOString().split('T')[0] + ' at ' + new Date(date).toISOString().split('T')[1].substring(0, 5)
