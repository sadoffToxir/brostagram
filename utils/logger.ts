import { WritableComputedRef, Ref } from 'vue'

export default function logger (consoleText: string, observed: WritableComputedRef<any> | Ref<any>) {
  const config = useRuntimeConfig()

  if (config.public.environment !== 'production') {
    watch(observed, useDebounce(() => {
    // eslint-disable-next-line no-console
      console.log(`${consoleText}:`, useCloneDeep(observed.value))
    }, 500), { deep: true })
  }
}
