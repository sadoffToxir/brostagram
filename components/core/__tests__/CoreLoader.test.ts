import { describe, it, expect } from 'vitest'
import { createTestingPinia } from '@pinia/testing'
import CoreLoader from '../CoreLoader.vue'
import { renderWithVuetify, screen } from '@/tests/vueTestingLibrary/utils'

const componentTag = 'CoreLoader'

describe('CoreLoader', () => {
  const setup = (isLoading = true) => renderWithVuetify(CoreLoader, {
    global: {
      plugins: [createTestingPinia()],
      mocks: {
        storeLoader: {
          isLoading
        }
      }
    }
  })

  it('should render with isLoading: true', () => {
    setup()

    const element = screen.getByTestId(componentTag)

    expect(element).toBeInTheDocument()
    expect(element).toMatchSnapshot()
  })

  it('should not render with isLoading: false', () => {
    setup(false)

    const element = screen.queryByTestId(componentTag)

    expect(element).not.toBeInTheDocument()
  })
})
