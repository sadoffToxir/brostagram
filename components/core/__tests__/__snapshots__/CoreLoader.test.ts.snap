// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoreLoader > should render with isLoading: true 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  class="v-progress-circular v-progress-circular--indeterminate v-theme--light text-primary"
  data-test="CoreLoader"
  role="progressbar"
  style="width: 64px; height: 64px;"
>
  <svg
    style="transform: rotate(calc(-90deg + 0deg));"
    viewBox="0 0 42.666666666666664 42.666666666666664"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      class="v-progress-circular__underlay"
      cx="50%"
      cy="50%"
      fill="transparent"
      r="20"
      stroke-dasharray="125.66370614359172"
      stroke-dashoffset="0"
      stroke-width="2.6666666666666665"
    />
    <circle
      class="v-progress-circular__overlay"
      cx="50%"
      cy="50%"
      fill="transparent"
      r="20"
      stroke-dasharray="125.66370614359172"
      stroke-dashoffset="125.66370614359172px"
      stroke-width="2.6666666666666665"
    />
  </svg>
  <!---->
</div>
`;
